OHS-Hosting-Infrastructure/complaint-tracker
# Pagination

## Description of change

~Reinvents the wheel of pagination.~  NEVER MIND

Uses Pagy to figure out the pagination frontend stuff. Configure defaults using `pagy.rb` initializer.

This is how it works:
- `params[:page]` passes info to `Hses` class which sets offset for API request, which then creates an `ApiResponse` subclass instance, `ApiResponseCollection`
- `ApiResponseCollection` has a `count` method that returns the number of items in the total result set
  - Pagy takes the number of items and the current page and returns an array of pages to be displayed, where `1` is a generic page, `"1"` is the currently selected page, and `:gap` is a separator (like "...")
 - The view uses that array to figure out what links to display and what classes to stick on them
- Adds a bunch of tests

## Acceptance Criteria

Adds pagination to UI where appropriate with the following conditions:

- 1 page or no results  (display nothing)
- <= 3 pages (display all) (functionally the same as below)
- Greater than 3 pages, display 1st page, pad 2 pages on either side of selected + selected, then last page

## How to test

There are a number of unit tests written you can look through.
I recommend turning on real data just to make sure that it's jumping through results or opening up the complaints controller and putting in "100" or whatever instead of the `res.count` and you'll see some pretty sweet pagination.
Currently there is no mechanism in the fake data for multiple pages.

## Issue(s)

* closes https://github.com/OHS-Hosting-Infrastructure/complaint-tracker/issues/158


## Checklist

To be completed by the submitter:



## To the Reviewer

This project is using [Conventional Comments](https://conventionalcomments.org/) to give structure
and context to PR comments. Please use these labels in your comments.

* **praise:**
* **nitpick:**
* **suggestion:**
* **issue:**
* **question:**
* **thought:**
* **chore:**
